lib/routes/builds.js

```diff
@@ -337,6 +356,72 @@ module.exports = function (app) {
       }).on('error', next);
     });
 
+  app.routes.put('/builds/:pkg/:env?', auth, asynk(async(req, res) => {
+    // Assume no locales are supplied, so just built in en-US, and no artifactory needed
+    const payload = req.body;
+    const models = app.models;
+    const pkg = models.Package.fromPublish(payload);
+
+    // create version and package records in system
+    var version = pkg.version;
+    var name = pkg.name;
+
+    try {
+      await manager.addVersionRecord(name, version, payload);
+      await manager.addPackageRecord(pkg);
```

Comment:
Shouldn't this happen after a version was succesfully published? What if the code below errors, do we revert? E.g. should this be split out into a `getVersionRecord` and `createVersionRecord`?

Reply:
manager.addVersionRecord actually does the getVersionRecord first, and if there is one already throws an error, which should be caught by the catch statement and terminates with `res.status(errCode).json({ message: err.message });`. I think we have to add the version record before the normal build process which isn't being run is supposed to add those records?

Reply:
I agree that we need to add the version record. However I'm curious if should be called here https://github.com/godaddy/warehouse.ai/pull/85/files#diff-0fe419fa6db6713b7daaf6ac88f7ca98R421 rather then before all logic runs. If the functions in `async.series` fail for whatever reason we would end up with a dangling record no?